pyriemann/transferlearning.py

```diff
@@ -0,0 +1,147 @@
+    SSVEP-based BCI using Riemannian similarities between users", in 26th
+    European Signal Processing Conference (EUSIPCO), pp. 1685-1689. IEEE, 2018.
+
+    [2] S. Khazem, S. Chevallier, Q. Barthélemy, K. Haroun and C. Noûs,
+    "Minimizing Subject-dependent Calibration for BCI with Riemannian Transfer
+    Learning", in 10th International IEEE/EMBS Conference on Neural
+    Engineering (NER), pp. 523-526. IEEE, 2021.
+    """
+
+    def __init__(self, metric='riemann', L=0, n_jobs=1):
+        """Init."""
+        self.metric = metric
+        self.n_jobs = n_jobs
+        self.L = L
+
+        if isinstance(metric, str):
+            self.metric_mean = metric
+            self.metric_dist = metric
+
+        elif isinstance(metric, dict):
+            # check keys
+            for key in ['mean', 'distance']:
+                if key not in metric.keys():
+                    raise KeyError('metric must contain "mean" and "distance"')
+
+            self.metric_mean = metric['mean']
+            self.metric_dist = metric['distance']
+
+        else:
+            raise TypeError('metric must be dict or str')
+
+    def fit(self, X, y, X_source, y_source, sample_weight=None):
+        """Fit (estimates) the centroids.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            ndarray of SPD matrices from target subject
+        y : ndarray shape (n_matrices, 1)
+            labels corresponding to each trial of target subject
+        X_source : ndarray, shape (n_matrices, n_channels, n_channels)
+            ndarray of SPD matrices from source domain subjects
+        y_source : ndarray shape (n_matrices, 1)
+            labels corresponding to each trial.
+        sample_weight : None | ndarray shape (n_matrices, 1)
+            the weights of each sample from the domain. if None, each sample
+            is treated with equal weights.
+
+        Returns
+        -------
+        self : MDWM instance
+            The MDWM instance.
+        """
+
+        if set(y) != set(y_source):
+            raise Exception(f"classes in source domain must match classes in target \
+                domain. Classes in source are {np.unique(y_source)} while \
+                    classes in target are {np.unique(y)}")
+
+        self.classes_ = np.unique(y)
+
+        if sample_weight is None:
+            sample_weight = np.ones(X_source.shape[0])
+
+        if self.n_jobs == 1:
+            self.target_means_ = [
+                mean_covariance(X[y == ll], metric=self.metric_mean)
+                for ll in self.classes_]
+
+            self.domain_means_ = [
+                mean_covariance(
+                    X_source[y_source == ll],
+                    metric=self.metric_mean,
+                    sample_weight=sample_weight[y_source == ll]
+                    )
+                for ll in self.classes_]
+        else:
+            self.target_means_ = Parallel(n_jobs=self.n_jobs)(
```

Comment:
`target_means_ ` and `domain_means_` should be declared in attributes.

Reply:
ok